tests/PHPStan/Rules/Comparison/data/number-comparison-operators.php

```diff
@@ -23,3 +23,75 @@ function (int $i): void {
 		}
 	}
 };
+
+function () {
+	if (-1 < null) {}
+	if (0 < null) {}
+	if (1 < null) {}
+	if (true < null) {}
+	if (false < null) {}
+	if ('1' < null) {}
+	if ('0' <= null) {}
+};
+
+function () {
+	if (true > false) {}
+	if (true >= false) {}
+	if (true < false) {}
+	if (true <= false) {}
+	if (false > true) {}
+	if (false >= true) {}
+	if (false < true) {}
+	if (false <= true) {}
+};
+
+function () {
+	if ('foo' < 'bar') {}
+	if ('foo' <= 'bar') {}
+	if ('foo' > 'bar') {}
+	if ('foo' >= 'bar') {}
+};
+
+function () {
+	if (1.9 > 1) {}
+	if ('1.9' > 1) {}
+
+	if (1.9 > 2.1) {}
+	if (1.9 > 1.5) {}
+	if (1.9 < 2.1) {}
+	if (1.9 < 1.5) {}
+};
+
+function (int $a, int $b) {
+	if (($a === 17 || $a === 42) && ($b === 3 || $b === 7)) {
+		if ($a < $b) {}
+		if ($a > $b) {}
+		if ($a <= $b) {}
+		if ($a >= $b) {}
+	}
+	if (($a === 11 || $a === 42) && ($b === 3 || $b === 11)) {
+		if ($a < $b) {}
+		if ($a > $b) {}
+		if ($a <= $b) {}
+		if ($a >= $b) {}
+	}
+};
+
+function (int $a, int $b) {
+	if ($a >= 10 && $a <= 20) {
+		if ($b >= 30 && $b <= 40) {
+			if ($a < $b) {}
+			if ($a > $b) {}
+			if ($a <= $b) {}
+			if ($a >= $b) {}
+		}
+	}
+	if ($a >= 10 && $a <= 25) {
+		if ($b >= 25 && $b <= 40) {
+			if ($a < $b) {}
+			if ($a > $b) {}
+			if ($a <= $b) {}
+			if ($a >= $b) {}
+		}
+	}
+};
```

Comment:
Hi, would you care to rewrite this as a new NodeScopeResolverTest::testFileAssert's dataProvider? It would be much clearer to see `assertType('false', 1.9 > 2.1)` instead of jumping between this file and rule test. Thanks!

Reply:
Sure thing.

Reply:
Done.